collector/collector.go

```diff
@@ -74,15 +75,15 @@ func (c *Collector) Start() error {
 			Description: "Lambda Collector",
 			Version:  Version,
 		},
-		ParserProvider: c.parserProvider,
+		ConfigMapProvider: c.mapProvider,
 		Factories:      c.factories,
 	}
 	var err error
 	c.svc, err = service.New(params)
 	if err != nil {
 		return err
 	}
-	cmd := service.NewCommand(c.svc)
+	cmd := service.NewCommand(params)
```

Comment:
@Aneurysm9 Could you please review this? Specifically I'd like to understand what to do with:

```
c.svc, err = service.New(params)
```

After this latest release. I think we can remove it, but it's used a couple times in this file.

Reply:
`service.NewCommand` no longer takes the service instance and instead uses the settings provided to create the  instance as part of `RunE`. https://github.com/open-telemetry/opentelemetry-collector/blob/release/v0.38.x/service/command.go#L34.

If you're using `service.NewCommand` as is, you can no longer use `service.New` to create the instance. It looks like you're using the service instance quite a bit, so you may need to create your own command instead of using `service.NewCommand`.

Reply:
Had to do something similar in https://github.com/aws-observability/aws-otel-collector/blob/main/cmd/awscollector/main.go#L108 to limit the flags.

Reply:
I see, so we can't access the service instance created by the latest `NewCommand` function?

Reply:
And assuming that's the case, do you think there's any reason we couldn't just copy the aws-otel-collector's `newCommand` implementation and use it here?

Reply:
I'd go the other direction.  Remove the use of `service.NewCommand()` since it doesn't do anything not already done here except set up some deprecated CLI flags.